config/muxcable.py

```diff
@@ -0,0 +1,175 @@
+            platform_sfputil.read_all_porttab_mappings(hwsku_path, multi_asic.get_num_asics())
+        else:
+            # For single ASIC platforms we pass port_config_file_path and the asic_inst as 0
+            port_config_file_path = device_info.get_path_to_port_config_file()
+            platform_sfputil.read_porttab_mappings(port_config_file_path, 0)
+    except Exception as e:
+        log.log_error("Error reading port info (%s)" % str(e), True)
+        sys.exit(3)
+
+    return
+
+# 'muxcable' command ("config muxcable mode <port|all> active|auto")
+@muxcable.command()
+@click.argument('port', metavar='<port_name>', required= True, default = None)
+@click.argument('state', metavar='<operation_status>', required=True, type=click.Choice(["active","auto"]))
+@click.option('--json','json_flag', required=False, is_flag=True, type=click.BOOL)
+def mode(state, port, json_flag):
+    """Show muxcable summary information"""
+
+    config_db = {}
+    state_db = {}
+    y_cable_tbl = {}
+    port_table_keys = {}
+
+    # Getting all front asic namespace and correspding config and state DB connector
+
+    namespaces = multi_asic.get_front_end_namespaces()
+    for namespace in namespaces:
+        asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+        config_db[asic_id] = swsscommon.DBConnector("CONFIG_DB", REDIS_TIMEOUT_MSECS, True, namespace)
+        state_db[asic_id] = swsscommon.DBConnector("STATE_DB", REDIS_TIMEOUT_MSECS, True, namespace)
+        #replace these with correct macros
+        y_cable_tbl[asic_id] = swsscommon.Table(state_db[asic_id], "MUX_CABLE_TABLE")
+        y_cable_update_tbl[asic_id] = swsscommon.Table(config_db[asic_id], "MUX_CABLE")
+        port_table_keys[asic_id] = y_cable_tbl[asic_id].getKeys()
+
+
+    if port is not None and port != "all":
+        asic_index = platform_sfputil.get_asic_id_for_logical_port(port)
+        click.echo("asic_index {} \n".format(asic_index))
+        if asic_index is None:
+            click.echo("Got invalid asic index for port {}, cant retreive mux status".format(port))
+            return
+
+        y_cable_asic_table = y_cable_tbl.get(asic_index, None)
+        if y_cable_asic_table is not None:
+            y_cable_asic_table_keys = y_cable_asic_table.getKeys()
+            if port in y_cable_asic_table_keys:
+                port_status_dict = {}
+                (status, fvs) = y_cable_asic_table.get(str(port))
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(port))
+                fvp = dict(fvs)
+                state_value = fvp.get("state",None)
+                if (state == "active" and state_value == "active") or (state == "auto" and state_value == "active"):
+                    # status is already active, so right back ok
+                    fvs = swsscommon.FieldValuePairs([('status', 'Ok')])
+                    y_cable_update_tbl[asic_index].set(logical_port_name, fvs)
```

Comment:
What is the purpose of those status fields?

Reply:
changed this to state